src/tasks/install-packages.ts

```diff
@@ -1,8 +1,54 @@
-import { projectInstall } from "pkg-install";
+import { execa } from "execa";
 
-export function installPackages(targetDir: string) {
-  return projectInstall({
-    cwd: targetDir,
-    prefer: "yarn",
-  });
+const INITIAL_PHASE_MAX = 70;
+const LINK_PHASE_MAX = 95;
+const UPDATE_INTERVAL = 250;
+
+export async function installPackages(targetDir: string, task: { output: string }): Promise<void> {
+  let progress = 0;
+  let linkPhaseStarted = false;
+
+  const updateProgress = () => {
+    const barLength = 30;
+    const filledLength = Math.floor((progress / 100) * barLength);
+    const bar = "█".repeat(filledLength) + "░".repeat(barLength - filledLength);
+    task.output = `Installing packages [${bar}] ${Math.round(progress)}%`;
+  };
+
+  const interval = setInterval(() => {
+    if (!linkPhaseStarted) {
+      const increment = Math.max(0.1, (INITIAL_PHASE_MAX - progress) / 50);
+      progress = Math.min(progress + increment, INITIAL_PHASE_MAX);
+    } else {
+      const increment = Math.max(0.05, (LINK_PHASE_MAX - progress) / 30);
```

Comment:
nit: Why 0.05, not 0.1 for example? As I understand it will be max when progress LINK_PHASE_MAX - progress < 1.5 which is very rare. 

I'm asking because there are magic numbers and I don't know why they were chosen. Like 50 could be INITIAL_PHASE_MAX, 30 could be LINK_PHASE_MAX - INITIAL_PHASE MAX. As I understand you configured them to make progress bar smooth?